nuxeo-runtime/nuxeo-runtime-test/src/main/java/org/nuxeo/runtime/test/NXRuntimeTestCase.java

```diff
@@ -398,6 +412,74 @@ public RuntimeContext deployTestContrib(String bundle, URL contrib) throws Excep
         return ctx;
     }
 
+    @Override
+    public void deployPartial(String name, Set<TargetExtensions> targetExtensions) throws Exception {
+        // Do not install bundle; we only need the Object to list his components
+        Bundle bundle = new BundleImpl(osgi, lookupBundle(name), null);
+
+        RuntimeContext ctx = new OSGiRuntimeContext(runtime, bundle);
+        listBundleComponents(bundle).stream().map(URLStreamRef::new).forEach(component -> {
+            try {
+                this.deployPartialComponent(ctx, targetExtensions, component);
+            } catch (IOException e) {
+                log.warn("PartialBundle: " + name + " failed to load: " + component, e);
+            }
+        });
+    }
+
+    /**
+     * Read a component from his StreamRef and create a new component (suffixed with `-partial`, and the base component
+     * name aliased) with only matching contributions of the extensionPoints parameter.
+     * 
+     * @param ctx RuntimeContext in which the new component will be deployed
+     * @param extensionPoints Set of Pair&lt;target, point&gt; that will be filtered
+     * @param component Reference to the original component
+     */
+    protected void deployPartialComponent(RuntimeContext ctx, Set<TargetExtensions> extensionPoints,
+            StreamRef component) throws IOException {
+        RegistrationInfo ri = ((DefaultRuntimeContext) ctx).createRegistrationInfo(component);
+        String alias = ri.getName().getName();
+        String name = alias.concat("-partial");
+
+        // Flatten Target Extension Points
+        Set<Pair<String, String>> targets = extensionPoints.stream()
+                                                           .map(TargetExtensions::getTargetExtensions)
+                                                           .flatMap(Set::stream)
+                                                           .collect(Collectors.toSet());
+
+        String ext = Arrays.stream(ri.getExtensions())
+                           .filter(e -> targets.contains(
+                                   Pair.of(e.getTargetComponent().getName(), e.getExtensionPoint())))
+                           .map(Extension::toXML)
+                           .reduce((o, x) -> o + x)
```

Comment:
You can use `collect(Collectors.joining())` here to avoid the lambda expression and the `orElse`.